pkg/build/generator/generator.go

```diff
@@ -235,6 +236,11 @@ func (g *BuildGenerator) Instantiate(ctx kapi.Context, request *buildapi.BuildRe
 		return nil, err
 	}
 
+	// add labels and annotations from the buildrequest.  existing label/annotations will take
+	// precedence because we don't want system annotations/labels (eg buildname) to get stomped on.
+	newBuild.Annotations = policy.MergeMaps(request.Annotations, newBuild.Annotations)
```

Comment:
@bparees does this also mean you can override the build config name annotation using this? that does not sound secure... Can we blacklist our API annotations?